lib/Target/ExportSMTLIB/ExportSMTLIB.cpp

```diff
@@ -0,0 +1,646 @@
+    return success();
+  }
+
+  LogicalResult visitSMTOp(IntConstantOp op, mlir::raw_indented_ostream &stream,
+                           ValueMap &valueMap) {
+    SmallString<16> str;
+    op.getValue().toStringSigned(str);
+    valueMap.insert(op.getResult(), str.str().str());
+    return success();
+  }
+
+  LogicalResult visitSMTOp(DeclareFunOp op, mlir::raw_indented_ostream &stream,
+                           ValueMap &valueMap) {
+    StringRef name =
+        names.newName(op.getNamePrefix() ? *op.getNamePrefix() : "tmp");
+    valueMap.insert(op.getResult(), name.str());
+    stream << "("
+           << (isa<SMTFuncType>(op.getType()) ? "declare-fun "
+                                              : "declare-const ")
+           << name << " ";
+    typeVisitor.dispatchSMTTypeVisitor(op.getType(), stream);
+    stream << ")\n";
+    return success();
+  }
+
+  LogicalResult visitSMTOp(AssertOp op, mlir::raw_indented_ostream &stream,
+                           ValueMap &valueMap) {
+    llvm::ScopedHashTableScope<Value, std::string> scope1(valueMap);
+    SmallVector<Value> worklist;
+    worklist.push_back(op.getInput());
+    stream << "(assert ";
+    VisitorInfo info(stream, valueMap, 8, 0);
+    if (failed(exprVisitor.printExpression(worklist, info)))
+      return failure();
+    stream << valueMap.lookup(op.getInput());
+    for (unsigned i = 0; i < info.openParens + 1; ++i)
+      stream << ")";
+    stream << "\n";
+    stream.indent(0);
+    return success();
+  }
+
+  LogicalResult visitSMTOp(CheckOp op, mlir::raw_indented_ostream &stream,
+                           ValueMap &valueMap) {
+    if (op->getNumResults() != 0)
+      return op.emitError() << "must not have any result values";
+
+    if (op.getSatRegion().front().getOperations().size() != 1)
+      return op->emitError() << "'sat' region must be empty";
+    if (op.getUnknownRegion().front().getOperations().size() != 1)
+      return op->emitError() << "'unknown' region must be empty";
+    if (op.getUnsatRegion().front().getOperations().size() != 1)
+      return op->emitError() << "'unsat' region must be empty";
+
+    stream << "(check-sat)\n";
+    return success();
+  }
+
+  LogicalResult visitUnhandledSMTOp(Operation *op,
+                                    mlir::raw_indented_ostream &stream,
+                                    ValueMap &valueMap) {
+    // Ignore operations which are handled in the Expression Visitor.
+    return success();
+  }
+
+private:
+  // A reference to the emission options for easy use in the visitor methods.
+  [[maybe_unused]] const SMTEmissionOptions &options;
+  TypeVisitor typeVisitor;
+  Namespace &names;
+  ExpressionVisitor exprVisitor;
+};
+
+} // namespace
+
+//===----------------------------------------------------------------------===//
+// Unified and Split Emitter implementation
+//===----------------------------------------------------------------------===//
```

Comment:
Does this really do unified or split emission? Looks like just unified emission at first glance.

Reply:
Good catch. Copy-pasted from another pass and forgot to remove this.